freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserService.java

```diff
@@ -112,39 +123,60 @@ private void asyncSynchronizeUsers(
         Set<String> userCrnFilter, Set<String> machineUserCrnFilter) {
         try {
             Set<String> environmentCrns = stacks.stream().map(Stack::getEnvironmentCrn).collect(Collectors.toSet());
+
+            Optional<String> requestIdOptional = Optional.of(
+                    Optional.ofNullable(MDCBuilder.getMdcContextMap().get(LoggerContextKey.REQUEST_ID.toString()))
+                            .orElseGet(() -> {
+                                String requestId = UUID.randomUUID().toString();
+                                LOGGER.debug("No requestId found. Setting request id to new UUID [{}]", requestId);
+                                MDCBuilder.addRequestId(requestId);
+                                return requestId;
+                            }));
+
+            Json umsEventGenerationIdsJson = userCrnFilter.isEmpty() && machineUserCrnFilter.isEmpty() ?
```

Comment:
I have a preference to have a variable isFullSync so you can say something like isFullSync ? ... : ....

Reply:
done